### PR TITLE
Fix optional in string interpolation

### PR DESCRIPTION
**Pull request by naftalibeder**

### CODE REVIEW

File: ios/RCTMGL-v10/RCTMGLLayer.swift [hunk 1]
```diff
@@ -219,7 +219,7 @@ class RCTMGLLayer : UIView, RCTMGLMapComponent, RCTMGLSourceConsumer {
         let decodedExpression = try JSONDecoder().decode(Expression.self, from: data)
         layer.filter = decodedExpression
       } catch {
-        Logger.log(level: .error, message: "parsing filters failed for layer \(optional: id): \(error.localizedDescription)")
```
Comment (mfazekas) [inline ios/RCTMGL-v10/RCTMGLLayer.swift:222]: What's wrong with `optional:`?!

Comment (naftalibeder) [reply ios/RCTMGL-v10/RCTMGLLayer.swift:222]: It's a little confusing to me, but it doesn't appear to be valid Swift (I don't even understand why it compiles!). I looked through the documentation ([String](https://developer.apple.com/documentation/swift/string) and [print()](https://developer.apple.com/documentation/swift/1541053-print)) and I don't see any syntax like that referenced.

image

In either case, it seems to force-unwrap the value, meaning the `log` call itself crashes the app if `id` is `nil`.

Comment (mfazekas) [reply ios/RCTMGL-v10/RCTMGLLayer.swift:222]: It's defined here:

https://github.com/rnmapbox/maps/blob/1049ab76957994ccfc837d6d53a8b1dc800c0f3f/ios/RCTMGL-v10/DefaultStringInterpolation%2Boptional.swift#L1-L5

And there is no force unwrap so it should not crash, just a shorter syntax as I find the `String(describing:...)` syntax verbose and ugly.

Comment (naftalibeder) [reply ios/RCTMGL-v10/RCTMGLLayer.swift:222]: Ha, ok! My mistake.

I did get a crash pointing to one of those `Logger.log()` calls indicating that the value was `nil`, but I'll just keep an eye out for whether I can reproduce this.

**Pull request closed without merging**